lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -2629,6 +2629,181 @@ void mme_app_handle_nw_init_ded_bearer_actv_req(
+  itti_s11_nw_init_deactv_bearer_request_t
+  *const nw_init_bearer_deactv_req_p)
+{
+  ue_mm_context_t *ue_context_p = NULL;
+  uint32_t i = 0;
+  OAILOG_FUNC_IN(LOG_MME_APP);
+  MessageDef *message_p = NULL;
+
+  for (i = 0; i< nw_init_bearer_deactv_req_p->no_of_bearers; i++) {
+    OAILOG_INFO(
+      LOG_MME_APP,
+      "Received Dedicated bearer deactivation Request from SGW with EBI %u\n",
+      nw_init_bearer_deactv_req_p->ebi[i]);
+  }
+  ue_context_p = mme_ue_context_exists_s11_teid(
+    &mme_app_desc.mme_ue_contexts, nw_init_bearer_deactv_req_p->s11_mme_teid);
+
+  if (ue_context_p == NULL) {
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+//Fetch PDN context
+  pdn_cid_t cid =
+  ue_context_p->bearer_contexts
+    [EBI_TO_INDEX(nw_init_bearer_deactv_req_p->ebi[0])]->pdn_cx_id;
+  pdn_context_t *pdn_context = ue_context_p->pdn_contexts[cid];
+
+  /* If delete_default_bearer is set and this is the only active PDN,
+  *  Send Detach Request to UE
+  */
+  if ((nw_init_bearer_deactv_req_p->delete_default_bearer) &&
+       (ue_context_p->nb_active_pdn_contexts == 1)) {
+    OAILOG_INFO(
+      LOG_MME_APP,
+      "Send detach to NAS for EBI %d as delete_default_bearer = true\n",
+      nw_init_bearer_deactv_req_p->ebi[0]);
+    //Send Deatch Request to NAS
+    if (ue_context_p->ecm_state == ECM_CONNECTED) {
+      mme_app_send_nas_detach_request(
+        ue_context_p->mme_ue_s1ap_id, MME_INITIATED_EPS_DETACH);
+    } else {
+      //If UE is in IDLE state send Paging Req
+      mme_app_paging_request_helper(
+        ue_context_p, true, false /* s-tmsi */, CN_DOMAIN_PS);
+      // Set the flag and send detach to UE after receiving service req
+      ue_context_p->emm_context.nw_init_bearer_deactv = true;
+    }
+  } else {
+    /*If UE is in connected state send Deactivate Bearer Req
+     * + ERAB Rel Cmd to NAS
+     */
+    if (ue_context_p->ecm_state == ECM_CONNECTED) {
+      message_p =
+      itti_alloc_new_message(TASK_MME_APP, MME_APP_DELETE_DEDICATED_BEARER_REQ);
+      if (message_p == NULL) {
+        OAILOG_INFO(
+        LOG_MME_APP,
+        "itti_alloc_new_message failed for"
+        "MME_APP_DELETE_DEDICATED_BEARER_REQ\n");
+        OAILOG_FUNC_OUT(LOG_MME_APP);
+      }
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).delete_default_bearer =
+        nw_init_bearer_deactv_req_p->delete_default_bearer;
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).ue_id =
+        ue_context_p->mme_ue_s1ap_id;
+      MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).no_of_bearers =
+        nw_init_bearer_deactv_req_p->no_of_bearers;
+      memcpy(
+        &MME_APP_DELETE_DEDICATED_BEARER_REQ(message_p).ebi,
+        nw_init_bearer_deactv_req_p->ebi,
+        sizeof(ebi_t));
+      // Send MME_APP_DELETE_DEDICATED_BEARER_REQ to NAS
+      itti_send_msg_to_task(TASK_NAS_MME, INSTANCE_DEFAULT, message_p);
+    } else {
+      /* If UE is in IDLE state remove bearer context
+       * send Create Bearer Rsp to SPGW
```

Comment:
send delete dedicated bearer rsp to spgw

Reply:
done